libs/types/src/permissions.rs

```diff
@@ -130,10 +129,10 @@ pub struct TrancheInvestors<Now, MinDelay, TrancheId, Moment> {
 /// This here implements trait Properties.
 #[derive(Encode, Decode, TypeInfo, Clone, Eq, PartialEq, Debug)]
 pub struct PermissionRoles<Now, MinDelay, TrancheId, Moment = u64> {
-	pool_admin: PoolAdminRoles,
-	currency_admin: CurrencyAdminRoles,
-	permissioned_asset_holder: PermissionedCurrencyHolders<Now, MinDelay, Moment>,
-	tranche_investor: TrancheInvestors<Now, MinDelay, TrancheId, Moment>,
+	pub pool_admin: PoolAdminRoles,
+	pub currency_admin: CurrencyAdminRoles,
+	pub permissioned_asset_holder: PermissionedCurrencyHolders<Now, MinDelay, Moment>,
+	pub tranche_investor: TrancheInvestors<Now, MinDelay, TrancheId, Moment>,
```

Comment:
Why do we need them public? Not against it, just curious and wondering if we should make it an API with methods on the struct rather than making them public.

Reply:
It's all because we have `cfg-types` and `cfg-traits`. Inside `cfg-traits`, we depend on `cfg-types`. To prevent a circular dependency, we can't rely on `cfg-traits` inside `cfg-types`. Which moves the `impl traits` out of `types` and into `traits`. We need to test the `impl functions` inside `traits`.

Reply:
I don't know if my answer makes sense. I also don't like, but for how we currently structure everything, it's hard not to make them public (with testing and everything).

Reply:
At the current commit of main. There is solely a dev dependency that is not even used. I am not sure, where the issue came from. But I think we can agree, that `traits` is stand alone and types can depend on traits and implement them.